### PR TITLE
fix9046 iOS Label text alignment in RTL

### LINKED ISSUE #9046: [Bug] IOS Label text alignment in RTL
### Description
In RTL , Label text alignment is left only in iOS, it's working fine in android.

### Steps to Reproduce
if the Label
1.  HorizontalOptions is "FillAndExpand"    
2. HorizontalTextAlignment is "Start" 
3. And FlowDirection is RTL

### Expected Behavior

Text alignment should be align from Right.

### Actual Behavior

Text is aligned left.

### Basic Information

- Version with issue:  Xamarin.Forms 4.4.0.991265
- Last known good version:   Xamarin.Forms 4.2.0.910310
- IDE:  Visual Studio Community 2019 for Mac
            Version 8.3.11 (build 1)
- Platform Target Frameworks: 
  - iOS:  iOS: 13.3
  - Android: Android: 9 
  
- Affected Devices: iOS Devices

### Screenshots
Screen Shot 2019-12-30 at 8 59 11 AM

### LINKED ISSUE #9447: [Bug] IOS Label text alignment in RTL
### Description

In RTL, Arabic text for the Label is not working. in both FlowDirection (RTL or LTR) it is aligned in the left. it happens only in iOS.

### Steps to Reproduce

1. HorizontalOptions is "FillAndExpand"
2. HorizontalTextAlignment is "Start"
3.  FlowDirection is RTL
4. and the Label Text is Arabic
 

### Expected Behavior

In RTL Mode, Text should be aligned in the right.

### Actual Behavior

In both mode (RTL or LTR), Text is aligned in the left

### Basic Information

- Version with issue: Xamarin.Forms 4.4.0.991537
- Last known good version: Xamarin.Forms 4.2.0.910310
- IDE:  Version: 6.10.0.17 (Visual Studio Community)
             Version 8.4.4 (build 91)

- Platform Target Frameworks:  
  - iOS:  Version: 13.10.0.17 (Visual Studio Community)
  - Android: Version: 16.4.0.10

- Android Support Library Version: 
- Nuget Packages:
- Affected Devices: All iOS Devices

### LINKED ISSUE #9751: [Bug] Misalignment of multi line label when using RTL on IOS
### Description
Misalignment of multi line label when using RTL on IOS but works fine on android
### Steps to Reproduce

1. Create ContentPage with label 
2. change FlowDirection of page to RightToLeft 
3. set label HorizontalTextAlignment="Start" 
4. observe the result

### Expected Behavior
label Text aligned right (start)

### Actual Behavior
label Text aligned left (end)
### Basic Information

- Version with issue: All versions of IOS
- Last known good version: None
- IDE: Visual Studio 2019 Version 16.4.3
- Platform Target Frameworks: 
  - iOS:  all versions
- Nuget Packages: Xamarin.Forms version 4.5.0.356
- Affected Devices: All Iphone 

### Screenshots

image

image

### Reproduction Link

https://github.com/amr-moomen/LabelMisalignment

### Workaround
Custom render for label on IOS

### PR DESCRIPTION
**Pull request by yurkinh**
### Description of Change ###

In RTL , Label text alignment is left only in iOS

### Issues Resolved ### 

- fixes #9046 
- fixes #9447
- fixes #9751  (duplicates)

### Platforms Affected ### 

- iOS

### Behavioral/Visual Changes ###
Text is aligned according to the flow direction

### Before/After Screenshots ### 

RTLFix

### PR Checklist ###

- [ ] Targets the correct branch
- [ ] Tests are passing (or failures are unrelated)

### CODE REVIEW

File: Xamarin.Forms.Platform.iOS/Renderers/LabelRenderer.cs [hunk 1]
```diff
@@ -389,6 +389,11 @@ void UpdateCharacterSpacing()
 
 			if (Element?.TextType != TextType.Text)
 				return;
+
+			if (Element.CharacterSpacing == 0)
```
Comment (samhouts) [inline Xamarin.Forms.Platform.iOS/Renderers/LabelRenderer.cs:393]: I think this will make it so you can't reset CharacterSpacing to 0.

Comment (yurkinh) [reply Xamarin.Forms.Platform.iOS/Renderers/LabelRenderer.cs]: Removed this unnecessary check. Probably left after testing.

**Comment by memu8** [comment]
I pulled your code edits, and created a simple test app with a label where HorizontalOptions = "FillAndExpand", HorizontalTextAlignment = "Start", FlowDirection = RTL, with text in Arabic. However, the text still seems to be left-aligned rather than right-aligned.

**Comment by yurkinh** [comment]
@memu8 Could you upload your test project?
 Cause I have tested it on samples provided with bug #9046

**Comment by memu8** [comment]
This issue was fixed in this [PR](https://github.com/xamarin/Xamarin.Forms/pull/10046/files#diff-1d0372a8878599126d4c9819fabb68eb) which is why when I replicated it, the fixes undid each other and I saw the text left-aligned. If you update to the latest Xamarin.Forms 4.6.0 version or later, the issue should be fixed. So sorry for the confusion!

**Pull request closed without merging**